product_code/matriculation_service/impl/src/test/scala/de/upb/cs/uc4/matriculation/impl/MatriculationServiceSpec.scala

```diff
@@ -305,6 +306,79 @@ class MatriculationServiceSpec extends AsyncWordSpec
+      val message = createSingleMatriculation(examReg0.name, "SS2020")
+      certificate.setup(student0.username)
+      certificate.getEnrollmentId(student0.username).invoke().flatMap { jsonId =>
+        client.getMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username + "thisShouldFail"))
+          .invoke(message).failed.map { answer =>
+            answer.asInstanceOf[UC4Exception].possibleErrorResponse.`type` should ===(ErrorType.OwnerMismatch)
+          }.andThen {
+            case _ => cleanup()
+          }
+      }
+    }
+
+    "fail submitting a proposal for adding matriculation data for a student, given a proposal while using another auth username" in {
+      val message = createSingleMatriculation(examReg0.name, "SS2020")
+      certificate.setup(student0.username)
+      certificate.getEnrollmentId(student0.username).invoke().flatMap { jsonId =>
+        client.getMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username))
+          .invoke(message).flatMap { proposal =>
+            client.submitMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username + "thisShouldFail"))
+              .invoke(SignedProposal(proposal.unsignedProposal, "c2lnbmVk")).failed.map { answer =>
+                answer.asInstanceOf[UC4Exception].possibleErrorResponse.`type` should ===(ErrorType.OwnerMismatch)
+              }.andThen {
+                case _ => cleanup()
+              }
+          }
+      }
+    }
+
+    "fail submitting a transaction for adding matriculation data for a student, given a proposal while using another auth username" in {
+      certificate.setup(student0.username)
+      certificate.getEnrollmentId(student0.username).invoke().flatMap { jsonId =>
+        prepare(Seq(
+          ImmatriculationData(
+            jsonId.id,
+            Seq(SubjectMatriculation(examReg0.name, Seq("SS2020")))
+          )
+        ))
+        client.getMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username))
+          .invoke(createSingleMatriculation(examReg0.name, "WS2020/21")).flatMap {
+            proposal =>
+
+              client.submitMatriculationProposal(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username))
+                .invoke(SignedProposal(proposal.unsignedProposal, "c2lnbmVk")).flatMap { transaction =>
+
+                  client.submitMatriculationTransaction(student0.username).handleRequestHeader(addAuthorizationHeader(student0.username + "thisShouldFail"))
+                    .invoke(SignedTransaction(transaction.unsignedTransaction, "c2lnbmVk")).failed.map(answer =>
+                      answer.asInstanceOf[UC4Exception].possibleErrorResponse.`type` should ===(ErrorType.OwnerMismatch))
+                }
+          }.andThen {
+            case _ => cleanup()
+          }
+      }
+    }
```

Comment:
```suggestion

```
To check for an OwnerMismatch, we do not look into the proposal at all; we just compare the authentication username with the username in the path. Since the check does not add any security at all, we removed it in operations. Therefore, these tests have to be deleted eventually.